src/main/java/ca/on/oicr/pineryreports/reports/impl/PreciseIdentityReport.java

```diff
@@ -0,0 +1,233 @@
+            .stream()
+            .filter(s -> s.getSampleType().equals(SAMPLE_CATEGORY_IDENTITY))
+            .collect(Collectors.toList());
+
+    table = new LinkedList<>();
+
+    // Add headers
+    table.add(
+        Lists.newArrayList(
+            "ID",
+            "Screening/Randomization Serum",
+            "Screening/Randomization Plasma",
+            "Screening/Randomization Buffy Coat",
+            "Screening/Randomization Urine Supernatant",
+            "Screening/Randomization Urine Pellet",
+            "6 mo Follow Up Serum",
+            "6 mo Follow Up Plasma",
+            "6 mo Follow Up Buffy Coat",
+            "1 yr Follow Up Serum",
+            "1 yr Follow Up Plasma",
+            "1 yr Follow Up Buffy Coat",
+            "1 yr Follow Up Urine Supernatant",
+            "1 yr Follow Up Urine Pellet",
+            "18 mo Follow Up Serum",
+            "18 mo Follow Up Plasma",
+            "18 mo Follow Up Buffy Coat",
+            "2 yr Follow Up Serum",
+            "2 yr Follow Up Plasma",
+            "2 yr Follow Up Buffy Coat",
+            "2 yr Follow Up Urine Supernatant",
+            "2 yr Follow Up Urine Pellet",
+            "Bx Slides - Positive",
+            "Bx Slides - Negative",
+            "RP Slides"));
+
+    // One row per identity
+    for (SampleDto identity : allPreciseIdentities) {
+      List<String> row = new LinkedList<>();
+      row.add(identity.getName());
+
+      // For all regular time points
+      for (int i = 0; i < 5; i++) {
+        TimePoint currentTimePoint = TimePoint.values()[i];
+        List<SampleDto> samplesAtCurrentTime =
+            allPreciseSamples
+                .stream()
+                .filter(
+                    s ->
+                        s.getName().startsWith(identity.getName())) // Get children of this Identity
+                .filter(currentTimePoint.predicate()) // Get samples at this time point
+                .collect(Collectors.toList());
+        // on 0th, 2th, 4th iterations use all the types, otherwise just 3
+        List<String> tissueOrigins;
+        if (i % 2 == 0) {
+          tissueOrigins = tissueOriginOrderExtra;
+        } else {
+          tissueOrigins = tissueOriginOrder;
+        }
+
+        // for all applicable tissue origins, get counts and add them to the row
+        for (String origin : tissueOrigins) {
+          row.add(
+              String.valueOf(
+                  samplesAtCurrentTime.stream().filter(s -> s.getName().contains(origin)).count()));
+        }
+      }
+
+      // Handle irregular time points
+      row.add(
+          String.valueOf(
+              allPreciseSamples
+                  .stream()
+                  .filter(s -> s.getName().startsWith(identity.getName()))
+                  .filter(s -> s.getSampleType().equals(SAMPLE_CLASS_SLIDE))
+                  .filter(TimePoint.BX.predicate())
+                  .filter(s -> true) // TODO: How to get Group ID? Need that for POS/NEG
```

Comment:
`SampleUtils#getAttribute` and/or `SampleUtils#getUpstreamAttribute`

Reply:
Updated